flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/FileExecutionGraphInfoStoreTest.java

```diff
@@ -341,9 +344,11 @@ public void testMaximumCapacity() throws IOException {
     /** Tests that a session cluster can terminate gracefully when jobs are still running. */
     @Test
     public void testPutSuspendedJobOnClusterShutdown() throws Exception {
+        final Duration timeout = Duration.ofSeconds(5);
         try (final MiniCluster miniCluster =
                 new PersistingMiniCluster(new MiniClusterConfiguration.Builder().build())) {
             miniCluster.start();
+            CommonTestUtils.waitUntilCondition(miniCluster::isRunning, Deadline.fromNow(timeout));
```

Comment:
This change doesn't make sense; after `miniCluster.start()` the `isRunning` flag is _always_ true.

Reply:
Have you considered the possibility that miniCluster.start() maybe throws an exception (== cluster never starting properly), but this exception being masked by the issue during the shutdown?

Reply:
What you could instead wait for is that the vertex is in a state running (==a slot was assigned => RM was started).